Refresh.GameServer/Endpoints/ApiV3/LevelApiEndpoints.cs

```diff
@@ -174,4 +177,83 @@ public ApiOkResponse SetLevelAsOverrideByHash(RequestContext context, GameDataba
         
         return new ApiOkResponse();
     }
+
+
+    [ApiV3Endpoint("levels/id/{id}/relations"), MinimumRole(GameUserRole.Restricted)]
+    [DocSummary("Gets your relations to a level by it's ID")]
+    [DocError(typeof(ApiNotFoundError), ApiNotFoundError.LevelMissingErrorWhen)]
+    public ApiResponse<ApiGameLevelRelationsResponse> GetLevelRelationsOfUser(RequestContext context, GameDatabaseContext database, GameUser user,
+        [DocSummary("The ID of the level")] int id)
+    {
+        GameLevel? level = database.GetLevelById(id);
+        if (level == null) return ApiNotFoundError.LevelMissingError;
+
+        return new ApiGameLevelRelationsResponse
+        {
+            IsHearted = database.IsLevelFavouritedByUser(level, user),
+            IsQueued = database.IsLevelQueuedByUser(level, user),
+            MyPlaysCount = database.GetTotalPlaysForLevelByUser(level, user)
+        };
```

Comment:
This should be refactored into a new method, `GameDatabaseContext.GetLevelRelationsByUser(level, user)`, although we don't have much precedent for GameDatabaseContext using API types directly, and that feels a bit dirty, so maybe `ApiGameLevelRelationsResponse.FromDatabase(database, level, user)` would be better? @jvyden what's your thoughts on this?

Reply:
Database code should be responsible for database code.

API code should be responsible for API code.

I don't see the problem with this.

Reply:
Also:

> we don't have much precedent for GameDatabaseContext using API types directly

This is intentional, as I plan to remove database access code from `Refresh.GameServer` entirely and have it live in its own project file.